File: leetcode-submissions/73_Set_Matrix_Zeroes.ts
/**
 Do not return anything, modify matrix in-place instead.
 */
 
 function setcol(matrix:number[][],col:number){
for(let i:number=0;i<matrix.length;i++){
    if(matrix[i][col]!=0){
        matrix[i][col]=-10;
    }
}
 }
function setrow(matrix:number[][],row:number){
    for(let i:number=0;i<matrix[0].length;i++){
    if(matrix[row][i]!=0){
        matrix[row][i]=-10;
    }
}
 }
function setZeroes(matrix: number[][]): void {
    for(let i:number=0;i<matrix.length;i++){
        for(let j:number=0;j<matrix[0].length;j++){
            if(matrix[i][j]===0){
                setrow(matrix,i);
                setcol(matrix,j);
            }
        }
    }
    for(let i:number=0;i<matrix.length;i++){
        for(let j:number=0;j<matrix[0].length;j++){
            if(matrix[i][j]===-10){
                matrix[i][j]=0;
            }
        }
    }
};

File: leetcode-submissions/73_Set_Matrix_Zeroes.cpp
class Solution {
public:
void setrow(vector<vector<int>>&arr,int row){
    for(int j=0;j<arr[0].size();j++){
        if(arr[row][j]!=0){
            arr[row][j]=-1;
        }
    }
}
void setcol(vector<vector<int>>&arr,int col ){
    for(int i=0;i<arr.size();i++){
        if(arr[i][col]!=0){
            arr[i][col]=-1;
        }
    }
}
void setzeromatrix(vector<vector<int>>&arr){
    for(int i=0;i<arr.size();i++){
        for(int j=0;j<arr[i].size();j++){
            if(arr[i][j]==0){
                setrow(arr,i);
                setcol(arr,j);
            }
        }
    }
}
    void setZeroes(vector<vector<int>>& matrix) {
       setzeromatrix(matrix);
       for(int i=0;i<matrix.size();i++){
           for(int j=0;j<matrix[i].size();j++){
               if(matrix[i][j]==-1){
                   matrix[i][j]=0;
               }
           }
       }

    }
};